Installation issue: gdbm

On Fedora 32 (default gcc is 10.0.1), gdbm fails with duplicate symbols.

### Steps to reproduce the issue

With gcc 10.0.1 as the compiler:
```console
$ spack install gdbm
```

### Information on your system

* **Spack:** 0.14.2-895-9698907e7
* **Python:** 3.8.2
* **Platform:** linux-fedora32-skylake

### Additional information

(Note: This could be a change to the linker behavior with 10.0.1. The same code builds find with gcc 9.x)

It appears to me that this is an actual bug in the code. The gdbm build system is attempting to link the gdbmtool executable with libgdbmapp.a. In gdbmtool.c two literals are defined:
```c
char *parseopt_program_doc = N_("examine and/or modify a GDBM database");
char *parseopt_program_args = N_("DBFILE");
```
These variables are already defined in libgdbmapp.a, which leads to the following error:

==> 966613: Installing gdbm
==> Using cached archive: /home/bergen/devel/tuxfan/spack/var/spack/cache/_source-cache/archive/86/86e613527e5dba544e73208f42b78b7c022d4fa5a6d5498bf18c8d6f745b91dc.tar.gz
==> Staging archive: /tmp/bergen/spack-stage/spack-stage-gdbm-1.18.1-k2ber44lxykjet7qpiscc4jqfhoiml3m/gdbm-1.18.1.tar.gz
==> Created stage in /tmp/bergen/spack-stage/spack-stage-gdbm-1.18.1-k2ber44lxykjet7qpiscc4jqfhoiml3m
==> No patches needed for gdbm
==> 966613: gdbm: Building gdbm [AutotoolsPackage]
==> 966613: gdbm: Executing phase: 'autoreconf'
==> 966613: gdbm: Executing phase: 'configure'
==> 966613: gdbm: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j12'

2 errors found in build log:
     227      CCLD     gdbm_load
     228      CCLD     gdbm_dump
     **229    /usr/bin/ld: /usr/bin/ld: ./libgdbmapp.a(parseopt.o):(.bss+0x8): mu
            ltiple definition of `parseopt_program_args'; ./libgdbmapp.a(parseo
            pt.o):(.bss+0x8): multiple definition of `**parseopt_program_args**'; g
            dbm_dump.o:(.data+0x70): first defined here
     230    /usr/bin/ld: ./libgdbmapp.a(parseopt.o):(.bss+0x10): multiple defin
            ition of `**parseopt_program_doc**'; gdbm_dump.o:(.data+0x78): first de
            fined here
     231    gdbm_load.o:(.data+0x140): first defined here
     232    /usr/bin/ld: ./libgdbmapp.a(parseopt.o):(.bss+0x10): multiple defin
            ition of `parseopt_program_doc'; gdbm_load.o:(.data+0x148): first d
            efined here**
  >> 233    collect2: error: ld returned 1 exit status
  >> 234    collect2: error: ld returned 1 exit status
     235    make[3]: *** [Makefile:648: gdbm_dump] Error 1
     236    make[3]: *** Waiting for unfinished jobs....
     237    make[3]: *** [Makefile:652: gdbm_load] Error 1
     238    make[3]: Leaving directory '/tmp/bergen/spack-stage/spack-stage-gdb
            m-1.18.1-k2ber44lxykjet7qpiscc4jqfhoiml3m/spack-src/src'
     239    make[2]: *** [Makefile:499: all] Error 2
     240    make[2]: Leaving directory '/tmp/bergen/spack-stage/spack-stage-gdb
            m-1.18.1-k2ber44lxykjet7qpiscc4jqfhoiml3m/spack-src/src'

--- a/var/spack/repos/builtin/packages/gdbm/package.py
+++ b/var/spack/repos/builtin/packages/gdbm/package.py
@@ -28,6 +28,7 @@ class Gdbm(AutotoolsPackage, GNUMirrorPackage):
     depends_on("readline")
     patch('gdbm.patch', when='%gcc@10:')
     patch('gdbm.patch', when='%clang@11:')
+    patch('gdbm.patch', when='%cce@11:')
 
     def configure_args(self):
 

